tests/distributions/test_reshape.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+from torch.autograd import Variable
+
+from pyro.distributions.torch import Bernoulli
+
+
+@pytest.mark.parametrize('sample_dim,extra_event_dims',
+                         [(s, e) for s in range(4) for e in range(4 + s)])
```

Comment:
Not exactly related to this PR, but it seems like right now we do not place any constraints on `extra_event_dims`. Would it be the case that `extra_event_dims <= len(self.base_dist.batch_shape)`? I am guessing that it should at least be `<= sample_shape + self.base_dist.batch_shape`, but correct me if I am mistaken.

Reply:
Good point. The requirement is that `extra_event_dims < len(sample_shape + batch_shape)`. In particular it's fine if `extra_event_dims > len(batch_shape)`. I've added a check to this PR and added death tests to verify the check.